source/mainnet/smart-contracts/onboarding-guide-ethereum-developers/faq.rst

```diff
@@ -0,0 +1,760 @@
+    and click on the top row `Updated contract instance at address: <783,0>`. You will see additional information
+    about the smart contract address, name, the function entry point that was invoked,
+    the CCD amount that was sent to the function, and events that were logged by this smart contract function.
+
+    The below picture shows that one event was logged by the contract `<783,0>` and no event was logged by the contract `<782,0>`.
+
+    .. image:: ./images/onboarding_ethereum_developers_1.png
+        :width: 100 %
+
+    If several events are logged by one function entry point, the different events can be distinguished by their array index.
+    The below picture shows that four events were logged by an entry point in
+    `this execution chain <https://dashboard.testnet.concordium.com/lookup/7fcad417384d8e36fd2264d16d0ce1385860cdad711d17f7d6c12137c9cbab2e>`_.
+
+    .. image:: ./images/onboarding_ethereum_developers_2.png
+        :width: 100 %
+
+.. dropdown::  How can I compare/decode the event data that I see on the dashboard?
+
+    A smart contract developer is free to use their own way of encoding and
+    interpreting event log data when writing their own smart contracts.
+
+    The official Concordium smart contracts use the following standard:
+    Each event has a number tag then two 0s and the rest of the event data.
+
+    .. image:: ./images/onboarding_ethereum_developers_11.png
+        :width: 100 %
+
+    For example, the above image has an event number tag of `fd` (hex encoding) which is `15*16+13 = 253` in decimal.
+    This number tag corresponds to a `burn event <https://github.com/Concordium/concordium-rust-smart-contracts/blob/main/concordium-cis2/src/lib.rs#L53>`_
+    of a `Cis2` token.
+    `u8::MAX` is 255 in decimal and `u8::MAX-2` is 253 in decimal (the same value than in the image above).
+
+    .. code-block:: console
+
+        pub const BURN_EVENT_TAG: u8 = u8::MAX - 2;
+
+    This number tag is used to distinguish between the different types of events.
+
+    In contrast, Ethereum uses a 32-byte long hash as an event tag which is called the event signature.
+
+    .. code-block:: console
+
+        eventTag (Ethereum) = hash(Transfer(address, address, uint)).
+
+    A downside of using a hash is that you have to use more than one byte for the tag
+    to avoid getting collisions. Concordium can efficiently store the event tag in
+    1 byte compared to the 32 bytes used by Ethereum.
+
+    You can compare the rest of the event data by adding the below lines of code to your test cases
+    and adjusting the `MyEventParams` to the event object that you are using.
+
+    .. code-block:: console
+
+        let parameter = MyEventParams {
+            example_key1: value1,
+            example_key1: value2,
+        }
+        let parameter_bytes = to_bytes(&parameter);
+        println!("(:x?)",parameter_bytes);
```

Comment:
Are you sure this is correct?
First of, I think you need to use `{` and `}`, then I don't think `Vec<u8>` supports the hex formatter.
Lastly, I believe the hex formatter should have zero padding, if it should correspond to the output of the dashboard.

Reply:
I describe two different options now:

Option1: println!("{:02x?}",parameter_bytes);
Option2: println!("{}", hex::encode(&parameter_bytes)); (`hex = "0.4"` needs to be added to the `Cargo.toml` file)